retroshare-gui/src/gui/common/RSTextBrowser.cpp

```diff
@@ -267,25 +267,57 @@ bool RSTextBrowser::checkImage(QPoint pos, QString &imageStr)
 /**
  * @brief RSTextBrowser::anchorForPosition Replace anchorAt that doesn't works as expected.
  * @param pos Where to get anchor from text
- * @return anchor If text at pos is inside anchor, else empty string.
+ * @return anchor If certain text at pos is inside anchor, else empty string.
  */
 QString RSTextBrowser::anchorForPosition(const QPoint &pos) const
 {
-	QTextCursor cursor = cursorForPosition(pos);
-	cursor.select(QTextCursor::WordUnderCursor);
 	QString anchor = "";
-	if (!cursor.selectedText().isEmpty()){
-		QRegExp rx("<a name=\"(.*)\"",Qt::CaseSensitive, QRegExp::RegExp2);
-		rx.setMinimal(true);
-		QString sel = cursor.selection().toHtml();
-		QStringList anchors;
-		int pos=0;
-		while ((pos = rx.indexIn(sel,pos)) != -1) {
-			anchors << rx.cap(1);
-			pos += rx.matchedLength();
+	QTextCursor line_with_cursor = cursorForPosition(pos);
+	// if click below text browser, then last string get used for some reason
+	if (line_with_cursor.atEnd()) {
+		return anchor;
+	}
+	// keepanchor to select text from the start of a line to cursor position
+	line_with_cursor.movePosition(QTextCursor::StartOfLine, QTextCursor::KeepAnchor); 
+	if (!line_with_cursor.selectedText().isEmpty()) {
+		QString selection = line_with_cursor.selection().toHtml();
+		// std::cout << "RSTextBrowser::AnchorForPosition selection: " << selection.toStdString() << std::endl;
+		QRegExp name_regexp("<a name=\"(.*)\"", Qt::CaseSensitive, QRegExp::RegExp2);
+		name_regexp.setMinimal(true);
+		// XXX may[be] get changed with time. Skip:
+		// 95 - DOCTYPE stuff
+		// 73 - <html><head...
+		// 33 - p, li...
+		// 22 - <body>...
+		// 119 - <p style...
+		                        /* for click to get what we seeking for */
+		int max_allowed_position = 95 + 73 + 33 + 22 + 119; // =342

```

Comment:
Same here. All these absolute numbers without name are pretty dangerous. Plz consts, or defines.

Reply:
> All these absolute numbers without name are pretty dangerous.

Totally agreed. Hope, that my compiled version couldn't be hacked via them, sorry for worries.